### DISCUSSION TITLE
How to update JTLReporter on a standalone workplace

### DISCUSSION THREAD

**Discussion by MoritzSalein** [ludeknovy/jtl-reporter#119] [Q&A] [answered]
Hi,
is it possible to get an instruction how I can update my installation of JTLReporter on a standalone workplace with Docker Desktop?
I'm not familiar with updating the JtlReporter container with docker compose. After my steps, see below, I don't have access to my old project and my test runs anymore.
My current additional problem is, that I worked with the former feature branch of  version 4.0 (feature/v4).

My failed steps are:
- stopping all JTLReporter container
- git switch to the main repository of the JTLReporter and overriding all old local files
- docker compose pull
- docker compose up -d
- take connection to the web interface of JTLReporter
- now I've to set new user and password
- after login I haven't any project in the dashboard

Many thanks in advance.

**Comment by ludeknovy** [comment, marked as answer]
Hi @MoritzSalein.

Always make a backup of your DB first, eg: `docker exec -t your-db-container pg_dumpall -c -U postgres > dump_`date +%d-%m-%Y"_"%H_%M_%S`.sql`

1. shut the app down, `docker-compose down`
2. Update the docker-compose.yml file
3. Run `docker-compose pull`
3. Run `docker-compose up -d`

In the scenario you described, it looks like you deleted your DB. You are asked to create a new user only when there is none in the DB.

Do you have both `jtl_reporter` and `jtl_reporter_v4` folder in `./data`?

**Reply by MoritzSalein** [reply]
Hi @ludeknovy,

Ok, I try it as you descirbed.

After my steps I have both folders in my data directory.
Before I made my update steps, I had only a `jtl_reporter` folder.

Thank you.

**Reply by ludeknovy** [reply]
Yeah, so your old data should be in `jtl_reporter`.

Remove the `jtl_reporter_v4 ` and rename the `jtl_reporter` to `jtl_reporter_v4` (or change docker-compose file to use `jtl_reporter`). That should make it work hopefully.

I changed the folder later on, and you started using the v4 when it was still under development, so that's where the discrepancy came in.

**Reply by MoritzSalein** [reply]
Thank you for your first class support. 

I stopped my failed JTLReporter update and delete the `jtl_reporter_v4` folder and changed the `jtl_reporter` folder to `jtl_reporter_v4` folder. After restart I got my old data in the dashboard.

**Reply by ludeknovy** [reply]
Btw if you want to test the recent updates you requested, you need to use the `latest` tag instead of `v4.x.x` as they were not released yet, but please bear in mind that the `latest` tag contains significant changes related to user roles.
